nix-tools/lib/Cabal2Nix.hs

```diff
@@ -45,18 +39,16 @@ import Data.String (fromString, IsString)
 
 -- import Distribution.Types.GenericPackageDescription
 -- import Distribution.Types.PackageDescription
-import Distribution.Types.PackageId
 --import Distribution.Types.Condition
-import Distribution.Types.UnqualComponentName
 import Nix.Expr
 import Data.Fix(Fix(..))
 import Data.Text (Text)
 
-import Cabal2Nix.Util (quoted, selectOr, mkThrow)
+import Cabal2Nix.Util (quoted, selectOr)
 
 data Src
   = Path FilePath
-  | PrivateHackage String
+  | Repo String (Maybe String)
```

Comment:
This is generalised to include hackage too. We treat all repositories in the same way.